lesson15/main.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+const wrapper = document.getElementById("js-wrapper");
+const ul = document.getElementById("js-ul");
+const submitBtn = document.getElementById("js-submitBtn");
+submitBtn.classList.add("btn");
+const modal = document.getElementById("js-modal");
+const openModalBtn = document.getElementById("js-openModalBtn");
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+const removeModal = () => {
+   modal.remove();
+   wrapper.classList.remove("bg_gray");
+};
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+openModalBtn.addEventListener("click", function () {
+   modal.style.display = "block";
+   wrapper.classList.add("bg_gray");
+   openModalBtn.remove();
+});
+
+const isEmptyString = (value) => value === "";
+
+submitBtn.addEventListener("click", () => {
+   const inputNumber = document.getElementById("js-number");
+   const inputText = document.getElementById("js-text");
+   if (isEmptyString(inputNumber.value) || isEmptyString(inputText.value)) { //inputNumber.valueとinputText.valueのデータ型がStringのため
+      alert("未入力です。");
+      return;
+   }
+   console.log('ID:' + inputNumber.value);
+   console.log('ユーザー名:' + inputText.value);
+   init();
+});
+
+async function init() {
+   removeModal();
+   startLoading();
+   try {
+      const result = await fetchData();
+      if (result.length === 0) {
+      throw new Error("データが空です");
+      }
+      createListElements(result);
+   } catch (e) {
+      const div = document.createElement("div");
+      ul.appendChild(div);
+      div.textContent = ` エラーが発生しました：${e}`;
+      console.log(e);
+   } finally {
+      endLoading();
+   }
+}
```

Comment:
IMO
If you use fetchData() & createListElements() in a try block, you will not know where the error is when it occurs.
Also, I wanted to emit loading only while fetching data, so I thought it would be better to separate the functions

Reply:
Thank you for your comment and advise.
I tried to fix that you pointed.
Please, check out and review. (I'm not sure about my code...)